pkg/apis/cloudcore/v1alpha1/types.go

```diff
@@ -0,0 +1,268 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	metaconfig "github.com/kubeedge/kubeedge/pkg/apis/meta/v1alpha1"
+)
+
+type CloudCoreConfig struct {
+	metav1.TypeMeta
+	// KubeAPIConfig set the kubernetes cluster info which cloudcore will connected
+	// +Required
+	KubeAPIConfig KubeAPIConfig `json:"kubeAPIConfig,omitempty"`
+	// Modules set cloudcore modules config
+	// +Required
+	Modules Modules `json:"modules,omitempty"`
+}
+
+type KubeAPIConfig struct {
+	// Master set the address of the Kubernetes API server (overrides any value in Kubeconfig)
+	// such as https://127.0.0.1:8443
+	// default ""
+	// Note: Can not use "omitempty" option,  It will affect the output of the default configuration file
+	Master string `json:"master"`
+	// ContentType set the ContentType of message transmission when interacting with k8s
+	// default application/vnd.kubernetes.protobuf
+	ContentType string `json:"contentType,omitempty"`
+	// QPS to use while talking with kubernetes apiserve
+	// default 100
+	QPS int32 `json:"qps,omitempty"`
+	// Burst to use while talking with kubernetes apiserver
+	// default 200
+	Burst int32 `json:"burst,omitempty"`
+	// Kubeconfig set path to kubeconfig file with authorization and master location information.
+	// default "/root/.kube/config"
+	KubeConfig string `json:"kubeConfig,omitempty"`
+}
+
+type Modules struct {
+	// CloudHub set cloudhub module config
+	CloudHub CloudHub `json:"cloudhub"`
+	// EdgeController set edgecontroller module config
+	EdgeController EdgeController `json:"edgecontroller"`
+	// DeviceController set devicecontroller module config
+	DeviceController DeviceController `json:"devicecontroller"`
+}
+
+type CloudHub struct {
+	// Enable set whether use this module, if false , need check other config
+	// default true
+	Enable bool `json:"enable"`
+	// KeepaliveInterval set keep alive interval (second)
+	// default 30
+	KeepaliveInterval int32 `json:"keepaliveInterval,omitempty"`
+	// NodeLimit set node limit
+	// default 10
+	NodeLimit int32 `json:"nodeLimit,omitempty"`
+	// TLSCAFile set ca file path
+	// default /etc/kubeedge/ca/rootCA.crt
+	TLSCAFile string `json:"tlsCAFile,omitempty"`
+	// TLSCertFile set cert file path
+	// default /etc/kubeedge/certs/edge.crt
+	TLSCertFile string `json:"tlsCertFile,omitempty"`
+	// TLSPrivateKeyFile set key file path
+	// default /etc/kubeedge/certs/edge.key
+	TLSPrivateKeyFile string `json:"tlsPrivateKeyFile,omitempty"`
+	// WriteTimeout set write time (second)
+	// default 30
+	WriteTimeout int32 `json:"writeTimeout,omitempty"`
+	// Quic set quic server info
+	// +Required
```

Comment:
Either Quic or websocket indicated is fine. So it's not required here.

Reply:
ok

Reply:
done